static/app/components/charts/optionSelector.tsx

```diff
@@ -33,24 +33,34 @@ function OptionSelector({
   multiple,
   ...rest
 }: SingleProps | MultipleProps) {
-  const mappedOptions = options.map(opt => ({
-    ...opt,
-    label: <Truncate value={String(opt.label)} maxLength={60} expandDirection="left" />,
-  }));
+  const mappedOptions = useMemo(() => {
+    return options.map(opt => ({
+      ...opt,
+      label: <Truncate value={String(opt.label)} maxLength={60} expandDirection="left" />,
+    }));
+  }, [options]);
 
   function onValueChange(option) {
-    if (multiple && option.length > 3) {
-      return;
-    }
     onChange(multiple ? option.map(o => o.value) : option.value);
   }
 
+  function isOptionDisabled(option) {
+    return (
+      // Option is explicitly marked as disabled
+      option.disabled ||
+      // The user has reached the maximum number of selections (3), and the option hasn't
+      // yet been selected. These options should be disabled to visually indicate that the
+      // user has reached the max.
+      (multiple && selected.length === 3 && !selected.includes(option.value))
+    );
+  }
+
```

Comment:
Users can only select up to 3 options in this menu. However, currently there is no visual indication that the remaining options (the 4th and 5th one in the screenshot below) are not selectable. This function marks those options as disabled so users are aware of that.

<img width="245" alt="Screen Shot 2022-05-10 at 12 40 50 PM" src="https://user-images.githubusercontent.com/44172267/167709021-27848c4d-fcc9-4614-a72f-7f411b818a62.png">